### DISCUSSION TITLE
web: Question (or possible bug) about duplicated `assets` prefix on audio_cache.dart

### DISCUSSION THREAD

**Discussion by jsnoriegam** [bluefireteam/audioplayers#1697] [General]
https://github.com/luanpotter/audioplayers/blob/ff7b82b87030f41bea300a9572c1ddebb0789a5b/packages/audioplayers/lib/src/audio_cache.dart#L107

Shouldn't this line be just:

`return Uri.parse('$prefix$fileName');`

Thanks.

**Comment by luanpotter** [comment]
good question, but I don't think so. I agree with you this will outcome to `assets/assets/...` but that is required on web for some reason. see how it makes the request on web:

![image](https://user-images.githubusercontent.com/882703/124175376-89a42280-da7b-11eb-8dba-329171146c4e.png)

I don't know why, but if you do a single "assets" it fails. basically all the flutter files are inside the first "assets" folder, all the compiled js, etc, in the same structure as the src code. which means there is an "assets" folder inside that, and that is where you assets are. for your understanding, the main.dart file would be found at `assets/lib/main.js` or something. so you can see where two assets come from. it is, indeed very weird but it's how flutter web works.

great question though!

**Comment by RelicOfTesla** [comment]
A long time....

**Comment by Gustl22** [comment]
All the source code of a flutter app is in a folder called `assets` which is not the same as the `assets` directory of the developer resources themselves. So they are by design nested.
See: https://github.com/flutter/engine/blob/2d39e672c95efc6c539d9b48b2cccc65df290cc4/lib/web_ui/lib/ui_web/src/ui_web/asset_manager.dart#L80C42-L80C42
And https://github.com/flutter/flutter/issues/67655
